Select single profile if only one profile is available else filter them for group

PatilShreyas opened the pull request:
Fixes: #496

gabrielittner commented:
Thank you for sending the pr. I missed it before I opened mine (#498) in which I split up the logic a bit more and also added a shared prefix case.

PatilShreyas commented:
> Thank you for sending the pr. I missed it before I opened mine (#498) in which I split up the logic a bit more and also added a shared prefix case.

Oh great!
